### PR TITLE
[addons] change binary interface (webbrowser required parts)

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description
The first 4 commits are include changes where required on the webbrowser.

Commits:
1. To include own keyboard GUI on browser addon, also must be the keypress there direct handled and not on end after a OK (the from Kodi where block the whole screen is not direct usable)
2. Only fix warnings where I see on windows
3. This add on the GUI action callback also the give of buttoncode and unicode where needed to handle actions on webbrowser web. Further brings it two functions to set control visible or to select
4. Is on audioengine where it can possible that audioengine is stopped and destructed and addon still try to call them, this check the engine is present before call
5. Only version increases, this is then also related to https://github.com/xbmc/xbmc/pull/16397 where a small piece on interface becomes changed

## Motivation and Context
To bring this hell of pain with webbrowser finally in Kodi :sweat_smile::smirk:

This is only a independent part of them and I hope to have the browser parts soon final.

## Screenshots (if appropriate):
Here about the keyboard and action calls:
Screenshot_20190724_234454

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h [hunk 1]
```diff
@@ -727,3 +752,124 @@ inline void KodiVersion(kodi_version_t& version)
 }
 } /* namespace kodi */
 //------------------------------------------------------------------------------
+
+//==============================================================================
+namespace kodi {
```
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h]: move curly bracket into a new line

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h [hunk 2]
```diff
@@ -727,3 +752,124 @@ inline void KodiVersion(kodi_version_t& version)
...
+      }
+    }
+  }
+  return ret;
+}
+} /* namespace kodi */
+//------------------------------------------------------------------------------
+
+//==============================================================================
+namespace kodi {
```
Comment (Rechi) [inline xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h]: move curly bracket into a new line

File: xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h (hunk 1 above)
Comment (AlwinEsch) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h]: Normally I would say yes and no problem, but I see it a little different with the addon heads.

Once the lowercase namespace to keep apart between Kodi page and addon page.
On the other hand, the `namespace kodi {` is added to each function, so when looking over the header an outsider sees directly what is needed.

If the namespace were just like in Kodi at the beginning of a file, the linebreak would make sense, but not there for me! If a line break is added to each namespace of a function or structure, the number of lines increases unnecessarily.

Comment (Rechi) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h]: Why is the `kodi` namespace closed and opened immediately afterwards again?
Please do use custom code style, without adding it to code guidelines and clang-format.

Comment (AlwinEsch) [reply xbmc/addons/kodi-addon-dev-kit/include/kodi/General.h]: One to see that it is an independent function and not within a class. The huge documentary can be damn long and could be overlooked.

The other is that under circumstances a "C" interface structure could come in between which should not be in a namespace.

In order to have the addon interface as inline it is not always possible to use the guidelines correctly.

**Pull request merged**